app/controllers/survey_controller.rb

```diff
@@ -15,9 +15,9 @@ def faqs
   def submit_survey
     demographic_data = DemographicData.new
     demographic_data.result = questionnaire_answers
-    demographic_data.person = person
+    demographic_data.person = current_user.person || @invite.person
```

Comment:
current_user is `nil` here, because user will be submitting survey without logging in.